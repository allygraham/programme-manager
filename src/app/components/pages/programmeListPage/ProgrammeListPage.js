import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import { constants } from '../../../constants';
import { Table } from '../../views/table';
import { TableRow } from '../../views/tableRow';
import { Search } from '../../views/search';
import { ProgrammeModal } from '../../pages/programmeModal';

const Title = styled('h2')`
  font-size: 2rem;
  margin-bottom: 1rem;
`;

const ButtonContainer = styled('div')`
  display: flex;
  justify-content: flex-end;
  margin-bottom: 2rem;
`;

const Notification = styled('article')`
  margin: 1rem 0;
`;

export const ProgrammeListPage = ({
  programmeList,
  openProgrammeModal,
}) => (
  <section className="section">
    <div className="container programme-list-page">
      <div className="columns">
        <div className="column is-8">
          <Title>{constants.titles.programmeList}</Title>
        </div>
        <div className="column is-4">
          <Search />
        </div>
      </div>
      <ButtonContainer>
        <button
          className="button is-primary"
          onClick={openProgrammeModal}
        >
          <span className="icon">
            <i className="fa fa-plus"></i>
          </span>
          <span>{constants.buttons.addProgramme}</span>
        </button>
      </ButtonContainer>
      {programmeList.length > 0 ?
        <Table>
          {programmeList.map(
            (programme, index) => 
              <TableRow
                programme={programme}
                key={index}
              />
          )}
        </Table> :
        <Notification className="message is-warning">
          <div className="message-body">
            There are no results for your search
          </div>
        </Notification>
      }
      <ProgrammeModal/>
    </div>
  </section>
);

ProgrammeListPage.propTypes = {
  programmeList: PropTypes.array.isRequired,
  openProgrammeModal: PropTypes.func.isRequired,
};
