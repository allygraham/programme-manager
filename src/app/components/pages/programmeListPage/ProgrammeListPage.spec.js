import React from 'react';
import { shallow } from 'enzyme';

import { ProgrammeListPage } from './ProgrammeListPage';

describe('ProgrammeListPage', () => {

  test('render the <ProgrammeListPage> component', () => {
    const programmeList = [{ id: 123 }, { id: 456 }];
    const openProgrammeModal = jest.fn();
    const isModalOpen = false;

    const wrapper = shallow(<ProgrammeListPage
      programmeList={programmeList}
      openProgrammeModal={openProgrammeModal}
      isModalOpen={isModalOpen}
    />);
    expect(wrapper).toMatchSnapshot();
  });

});
