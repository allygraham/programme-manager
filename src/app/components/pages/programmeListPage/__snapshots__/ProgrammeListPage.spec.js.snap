// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgrammeListPage render the <ProgrammeListPage> component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgrammeListPage
    isModalOpen={false}
    openProgrammeModal={[MockFunction]}
    programmeList={
      Array [
        Object {
          "id": 123,
        },
        Object {
          "id": 456,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container programme-list-page"
      >
        <div
          className="columns"
        >
          <div
            className="column is-8"
          >
            <Styled(h2)>
              Programme List
            </Styled(h2)>
          </div>
          <div
            className="column is-4"
          >
            <Connect(Search) />
          </div>
        </div>
        <Styled(div)>
          <button
            className="button is-primary"
            onClick={[MockFunction]}
          >
            <span
              className="icon"
            >
              <i
                className="fa fa-plus"
              />
            </span>
            <span>
              Add Programme
            </span>
          </button>
        </Styled(div)>
        <Table>
          <Connect(TableRow)
            programme={
              Object {
                "id": 123,
              }
            }
          />
          <Connect(TableRow)
            programme={
              Object {
                "id": 456,
              }
            }
          />
        </Table>
        <withRouter(Connect(ProgrammeModal)) />
      </div>,
      "className": "section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="columns"
          >
            <div
              className="column is-8"
            >
              <Styled(h2)>
                Programme List
              </Styled(h2)>
            </div>
            <div
              className="column is-4"
            >
              <Connect(Search) />
            </div>
          </div>,
          <Styled(div)>
            <button
              className="button is-primary"
              onClick={[MockFunction]}
            >
              <span
                className="icon"
              >
                <i
                  className="fa fa-plus"
                />
              </span>
              <span>
                Add Programme
              </span>
            </button>
          </Styled(div)>,
          <Table>
            <Connect(TableRow)
              programme={
                Object {
                  "id": 123,
                }
              }
            />
            <Connect(TableRow)
              programme={
                Object {
                  "id": 456,
                }
              }
            />
          </Table>,
          <withRouter(Connect(ProgrammeModal)) />,
        ],
        "className": "container programme-list-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="column is-8"
              >
                <Styled(h2)>
                  Programme List
                </Styled(h2)>
              </div>,
              <div
                className="column is-4"
              >
                <Connect(Search) />
              </div>,
            ],
            "className": "columns",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Styled(h2)>
                  Programme List
                </Styled(h2)>,
                "className": "column is-8",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Programme List",
                },
                "ref": null,
                "rendered": "Programme List",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Search) />,
                "className": "column is-4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="button is-primary"
              onClick={[MockFunction]}
            >
              <span
                className="icon"
              >
                <i
                  className="fa fa-plus"
                />
              </span>
              <span>
                Add Programme
              </span>
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="icon"
                >
                  <i
                    className="fa fa-plus"
                  />
                </span>,
                <span>
                  Add Programme
                </span>,
              ],
              "className": "button is-primary",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fa fa-plus"
                  />,
                  "className": "icon",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-plus",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Programme",
                },
                "ref": null,
                "rendered": "Add Programme",
                "type": "span",
              },
            ],
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Connect(TableRow)
                programme={
                  Object {
                    "id": 123,
                  }
                }
              />,
              <Connect(TableRow)
                programme={
                  Object {
                    "id": 456,
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "programme": Object {
                  "id": 123,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "programme": Object {
                  "id": 456,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container programme-list-page"
        >
          <div
            className="columns"
          >
            <div
              className="column is-8"
            >
              <Styled(h2)>
                Programme List
              </Styled(h2)>
            </div>
            <div
              className="column is-4"
            >
              <Connect(Search) />
            </div>
          </div>
          <Styled(div)>
            <button
              className="button is-primary"
              onClick={[MockFunction]}
            >
              <span
                className="icon"
              >
                <i
                  className="fa fa-plus"
                />
              </span>
              <span>
                Add Programme
              </span>
            </button>
          </Styled(div)>
          <Table>
            <Connect(TableRow)
              programme={
                Object {
                  "id": 123,
                }
              }
            />
            <Connect(TableRow)
              programme={
                Object {
                  "id": 456,
                }
              }
            />
          </Table>
          <withRouter(Connect(ProgrammeModal)) />
        </div>,
        "className": "section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="columns"
            >
              <div
                className="column is-8"
              >
                <Styled(h2)>
                  Programme List
                </Styled(h2)>
              </div>
              <div
                className="column is-4"
              >
                <Connect(Search) />
              </div>
            </div>,
            <Styled(div)>
              <button
                className="button is-primary"
                onClick={[MockFunction]}
              >
                <span
                  className="icon"
                >
                  <i
                    className="fa fa-plus"
                  />
                </span>
                <span>
                  Add Programme
                </span>
              </button>
            </Styled(div)>,
            <Table>
              <Connect(TableRow)
                programme={
                  Object {
                    "id": 123,
                  }
                }
              />
              <Connect(TableRow)
                programme={
                  Object {
                    "id": 456,
                  }
                }
              />
            </Table>,
            <withRouter(Connect(ProgrammeModal)) />,
          ],
          "className": "container programme-list-page",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="column is-8"
                >
                  <Styled(h2)>
                    Programme List
                  </Styled(h2)>
                </div>,
                <div
                  className="column is-4"
                >
                  <Connect(Search) />
                </div>,
              ],
              "className": "columns",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Styled(h2)>
                    Programme List
                  </Styled(h2)>,
                  "className": "column is-8",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Programme List",
                  },
                  "ref": null,
                  "rendered": "Programme List",
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(Search) />,
                  "className": "column is-4",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="button is-primary"
                onClick={[MockFunction]}
              >
                <span
                  className="icon"
                >
                  <i
                    className="fa fa-plus"
                  />
                </span>
                <span>
                  Add Programme
                </span>
              </button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="icon"
                  >
                    <i
                      className="fa fa-plus"
                    />
                  </span>,
                  <span>
                    Add Programme
                  </span>,
                ],
                "className": "button is-primary",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="fa fa-plus"
                    />,
                    "className": "icon",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-plus",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add Programme",
                  },
                  "ref": null,
                  "rendered": "Add Programme",
                  "type": "span",
                },
              ],
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Connect(TableRow)
                  programme={
                    Object {
                      "id": 123,
                    }
                  }
                />,
                <Connect(TableRow)
                  programme={
                    Object {
                      "id": 456,
                    }
                  }
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "programme": Object {
                    "id": 123,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "programme": Object {
                    "id": 456,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
